src/JsonPacker.php

```diff
@@ -28,10 +29,18 @@ public function __construct()
      * pack entity as a json string.
      *
      * @param Entity $entity
+     *
      * @return string
+     * @throws \UnexpectedValueException
      */
     public function pack(Entity $entity)
     {
-        return json_encode(array($entity->getTag(), $entity->getTime(), $entity->getData()));
+        $json = json_encode(array($entity->getTag(), $entity->getTime(), $entity->getData()));
+
+        if (!$json and json_last_error() !== JSON_ERROR_NONE) {
```

Comment:
why did you use "and" instead of "&&" in this statement? typo?

Reply:
Why: I don't like adding parentheses just to fix precedence.  "and" is a fully valid logic operator with lower precedence.